src/query/block/meta.go

```diff
@@ -158,6 +161,31 @@ func (m ResultMetadata) IsDefault() bool {
 	return m.Exhaustive && m.LocalOnly && len(m.Warnings) == 0
 }
 
+// VerifyTemporalRange will verify that each resolution seen is below the
+// given step size, adding warning headers if it is not.
+func (m *ResultMetadata) VerifyTemporalRange(step time.Duration) {
+	// NB: this map is unlikely to have more than 2 elements in real execution,
+	// since these correspond to namespace count.
+	invalidResolutions := make(map[time.Duration]struct{}, 10)
+	for _, res := range m.Resolutions {
+		if res > step {
+			invalidResolutions[res] = struct{}{}
```

Comment:
Do we really need this map and also the sorting? Probably simpler to just compile `[]string` warnings in this loop, and after `if len(warnings) > 0` then `m.AddWarning`